pkg/util/tls.go

```diff
@@ -0,0 +1,78 @@
+package util
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+
+	pb "github.com/buildbarn/bb-storage/pkg/proto/configuration/server"
+
+	"google.golang.org/grpc/credentials"
+)
+
+// ValidateTls checks params are in order, return non-nil if not.
+// must be called before using.
+func ValidateTls(p *pb.TLSConfiguration) error {
+	if p == nil {
+		return nil
+	}
+	if p.CertFile == "" || p.KeyFile == "" {
+		return errors.New("Must supply both cert file and key file if using TLS")
+	}
+	return nil
+}
+
+// MakeGrpcCreds Create TLS credentials for GRPC
+// return nil,nil to indicate no TLS if neither parameter is configured (empty string)
+func MakeGrpcCreds(tlsParams *pb.TLSConfiguration) (credentials.TransportCredentials, error) {
+	if tlsParams == nil {
+		return nil, nil
+	}
+	cert, err := tls.LoadX509KeyPair(tlsParams.CertFile, tlsParams.KeyFile)
+	if err != nil {
+		return nil, fmt.Errorf("Can't load X509 certificate and keypair with config %v: %s",
+			tlsParams, err)
+	}
+	cfg := &tls.Config{Certificates: []tls.Certificate{cert}}
+	creds := credentials.NewTLS(cfg)
+	return creds, nil
+}
+
+// several helpers to help with testing
```

Comment:
Not only testing right? Since this is used by HttpListenAndServe which is called in `main.go`

Reply:
Oh yes, both code paths use them, just that without the testing usage, the code here could have been a bit simpler. I can expand the comment.